.github/workflows/generate-baselines.yml

```diff
@@ -0,0 +1,113 @@
+name: Generate Baselines
+
+on:
+  workflow_dispatch:
+    inputs:
+      dummy:
+        description: 'Place holder'
+        required: false
+        default: 'Dummy argument'
+
+  push:
+    branches:
+      - development
+
+env:
+  AUTOMLBENCHMARK_REPO: 'openml/automlbenchmark'
+  AUTOMLBENCHMARK_REF: 'master'
+  GITHUB_URL: 'https://github.com'
+  GITHUB_RAW_URL: 'https://raw.githubusercontent.com'
+  FILE_SOURCE_DIR: '.github/workflows/benchmarking-files'
+
+jobs:
+
+  generate-baselines:
+    name: Generate Baselines
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        task_type: [regression, classification]
+
+    steps:
+      - name: Debug
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: |
+          echo "$GITHUB_CONTEXT"
+          pwd
+          ls -a
+
+      - name: Branch extract
+        id: extract
+        env:
+          head_ref: ${{ github.head_ref }}
+        run: |
+          # Sometimes a pull request does or doesn't have head_ref (branch) set.
+          # When it is not, GITHUB_REF holds the branch
+          if [[ -z $head_ref ]]
+          then
+            echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+          else
+            echo "##[set-output name=branch;]$(echo $head_ref)"
+          fi
+        # outputs
+        #   branch: the branch name
+
+      - name: Setup Python 3.9.6
```

Comment:
Not that I can think of other than to stick with the highest version we test against in the unittests?

Reply:
Which version would we get if you don't specify anything? That would basically check that we can work with what the latest Ubuntu provides, which is probably what a lot of people are using.

Reply:
It indicates it uses the latest `python 3.x`, which I assume does not include release candidates or beta versions of Python. In this case the latest full release is `3.9.6` with `3.10` still in development.

I think the version that ships with `ubuntu 20.04` is `3.8.6` but I'm not entirely sure.

Reply:
It's `Python 3.8.10` on my machine. I'd be in favor of using that.

Reply:
Cool, I've updated it to indicate Python `3.8` which should just be the latest `3.8.x` version.

Reply:
Ah, sorry, I meant to ask whether we can drop any version indication?

Reply:
If we drop version indication, it will default to `3.x` which translates to `3.9.6`

Reply:
Ah, I thought we would then get 3.8 as the Ubuntu default. Is there a way to get the Ubuntu default? But maybe this is also irrelevant...

Reply:
Looking at the [docs](https://github.com/actions/setup-python) and the [actions.yml](https://github.com/actions/setup-python/blob/main/action.yml#L7) it doesn't seem possible natively. However it could be possible to do something like the following.
```
python-version: $(python3 -V)
```

I have no idea if this would work, it may need to be put in a step before and output it as a variable but the same principal applies, just get the system version and then have the `setup-python` action set that version up for use.

Reply:
Also to add, whether it is irrelevant or not ... I think it is relevant to be consistent as Cython updates may increase efficiency which leads to better performances. We may notice a difference on Python version changes but I try to read them and there's about one or two optimizations per major version change.

Given that it should be consistent, I would agree that it makes sense to use `ubuntu-latest`

Reply:
Great, could you please do so?